packages/serverpod/lib/src/server/session.dart

```diff
@@ -146,6 +148,22 @@ abstract class Session {
     sessionLogs = server.serverpod.logManager.initializeSessionLog(this);
   }
 
+  LogWriter _createLogWriter(Session session, LogSettingsManager settings) {
+    var logSettings = settings.getLogSettingsForSession(session);
+
+    LogWriter logWriter = switch (Features.enablePersistentLogging) {
+      (true) => DatabaseLogWriter(session),
+      (false) => StdOutLogWriter(session),
+    };
+
+    if ((session is StreamingSession || session is MethodStreamSession) &&
```

Comment:
Question: Is this check duplicated in other places? Meaning certain attributes are given to StreamingSession based on their class names? If so, could we centralize it to a function?